protocol/communities_messenger_test.go

```diff
@@ -4225,3 +4225,37 @@ func (s *MessengerCommunitiesSuite) TestIsDisplayNameDupeOfCommunityMember() {
 	s.Require().NoError(err)
 	s.Require().False(result)
 }
+
+func (s *MessengerCommunitiesSuite) TestUpdateGrantsForControlledCommunities() {
+	grantUpdateInterval = 1 * time.Millisecond
+	communities.GrantExpirationTime = 10 * time.Millisecond
+
+	community, _ := createEncryptedCommunity(&s.Suite, s.owner)
+	s.Require().True(community.Encrypted())
+
+	s.owner.communitiesManager.PermissionChecker = &testPermissionChecker{}
+
+	s.advertiseCommunityTo(community, s.owner, s.alice)
+	s.joinCommunity(community, s.owner, s.alice)
+
+	initialGrant, _, err := s.alice.communitiesManager.GetCommunityGrant(community.IDString())
+	s.Require().NoError(err)
+	s.Require().NotEqual(initialGrant, []byte{})
+
+	time.Sleep(communities.GrantExpirationTime)
+	s.owner.updateGrantsForControlledCommunities()
+
+	// TODO: notify client about new grant received
```

Comment:
@osmaczko should i add updated grant to the `state.Response` or whatever event i should use on the receiver side for waiting?

Reply:
Depends if we need that on the client side. Do we have any UI indication of expired grant in community we are a member of?

PS. Having grant in resposne would make the test a bit cleaner, because we wouldn't need explicit `time.Sleep`.